#VRML_SIM R2022a utf8
WorldInfo {
  basicTimeStep 16
  contactProperties [
    ContactProperties {
      material2 "caster"
      coulombFriction [
        0.01
      ]
      softERP 0.1
      softCFM 0.0002
    }
  ]
}
Viewpoint {
  orientation 0.30073251839550724 0.1383781466244037 -0.9436161512587048 2.322321087018257
  position 15.756061683753055 8.00082207327904 11.849079513463744
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Solid {
  translation 3.1 -2.1 -0.1
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry DEF Floor Box {
        size 6 4 0.2
      }
    }
  ]
  name "floor"
  boundingObject USE Floor
}
Solid {
  translation 6.15 -2.1 1
  children [
    Solid {
      translation -3.05 0 1
      children [
        DEF Roof Shape {
          appearance PBRAppearance {
            transparency 1
          }
          geometry Box {
            size 6 4 0.1
          }
          castShadows FALSE
        }
      ]
      name "Roof"
      boundingObject USE Roof
    }
    Solid {
      translation -3.05 -2.05 0
      rotation 0 0 1 1.5708
      children [
        DEF LONG_WALL Shape {
          appearance DEF Wall_app PBRAppearance {
            transparency 0.6
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 6 2
          }
        }
      ]
      name "EastWall_l"
      boundingObject USE LONG_WALL
    }
    Solid {
      translation -6.1 0 0
      children [
        DEF SHORT_WALL Shape {
          appearance USE Wall_app
          geometry Box {
            size 0.1 4.2 2
          }
        }
      ]
      name "SouthWall_s"
      boundingObject USE SHORT_WALL
    }
    Solid {
      translation -3.05 2.05 0
      rotation 0 0 1 -1.5707953071795862
      children [
        DEF LONG_WALL Shape {
          appearance USE Wall_app
          geometry Box {
            size 0.1 6 2
          }
        }
      ]
      name "WestWall_l"
      boundingObject USE LONG_WALL
    }
    DEF SHORT_WALL Shape {
      appearance USE Wall_app
      geometry Box {
        size 0.1 4.2 2
      }
    }
  ]
  name "Walls"
  boundingObject USE SHORT_WALL
}
Solid {
  translation 0.36 -3.7 0.75
  children [
    Shape {
      appearance DEF cabinet PBRAppearance {
        baseColor 1 0.862745 0.862745
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.5 0.79 1.5
      }
    }
  ]
}
Solid {
  translation 0.36 -2.46 0.34
  children [
    Shape {
      appearance USE cabinet
      geometry Box {
        size 0.5 1.7 0.64
      }
    }
  ]
  name "solid(1)"
}
Solid {
  translation 0.2 -2.46 1.33
  children [
    Shape {
      appearance USE cabinet
      geometry Box {
        size 0.2 1.7 0.45
      }
    }
  ]
  name "solid(2)"
}
Solid {
  translation 4.5 -2.06 0
  rotation 0 0 1 3.14159
  children [
    DEF ct-scanner Transform {
      translation 0.94 2.41645e-06 0.44
      rotation 0 0 -0.9999999999999999 1.5707953071795862
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.4 1.6 0.18
          }
        }
      ]
    }
    DEF ct-scanner Transform {
      translation 1.09 2.56945e-06 0.18
      rotation 0 0 -0.9999999999999999 1.5707953071795862
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.654902 0.662745 0.682353
            roughness 1
            metalness 0.5
          }
          geometry Box {
            size 0.28 0.75 0.36
          }
        }
      ]
    }
    DEF ct-scanner Transform {
      translation -0.21 1.24381e-06 0.6
      rotation 0 0 -0.9999999999999999 1.5707953071795862
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.866667 0.870588 0.882353
            roughness 1
            metalness 0.5
          }
          geometry Box {
            size 1.4 0.53 1.17
          }
        }
      ]
    }
  ]
  name "ct-scaner"
  boundingObject USE ct-scanner
}
Chair {
  translation 0.4 -1.2 0
  rotation 0 0 1 -5.307179586466759e-06
  name "chair(1)"
  physics Physics {
  }
}
Chair {
  translation 0.4 -0.56 0
  rotation 0 0 1 -5.307179586466759e-06
  name "chair(2)"
  physics Physics {
  }
}
Solid {
  translation 0.95 -2.78 0.21
  children [
    Transform {
      translation 0.067615 -0.0181172 0.47
      rotation 0 0 1 -0.261795307179586
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.4 0.4 0.5
          }
        }
      ]
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0.298039 0.333333 0.427451
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.45
        radius 0.04
      }
    }
  ]
  name "solid(3)"
}
Robot {
  translation 2.17 -0.79 0.04
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.05 0.06 0
        rotation 1 0 0 1.5707996938995747
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "WHEEL1"
        boundingObject USE WHEEL
        physics DEF WHEEL_PY Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint DEF WHEEL2 Solid {
        translation 0.05 -0.06 0
        rotation 1 0 0 1.5707996938995747
        children [
          USE WHEEL
        ]
        name "WHEEL2"
        boundingObject USE WHEEL
        physics USE WHEEL_PY
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint DEF WHEEL3 Solid {
        translation -0.05 0.06 0
        rotation 1 1.1698080713731192e-24 0 1.5707996938995747
        children [
          USE WHEEL
        ]
        name "WHEEL3"
        boundingObject USE WHEEL
        physics USE WHEEL_PY
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.05 -0.06 0
        rotation 0.1256516415892658 -0.7015013724228277 0.7015037344553569 2.8916003679250695
        children [
          USE WHEEL
        ]
        name "WHEEL4"
        boundingObject USE WHEEL
        physics USE WHEEL_PY
      }
    }
    Solid {
      translation 0 0 0.175
      children [
        PointLight {
          attenuation 0 0 1
          color 0 0 1
          intensity 10
          location 0 0 0.25
          castShadows TRUE
        }
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry DEF POLE Cylinder {
            height 0.3
            radius 0.02
          }
        }
      ]
      name "Light Pole"
      boundingObject USE POLE
      physics Physics {
      }
    }
    GPS {
      translation 0.1 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
          }
          geometry DEF GPS_BOX Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "gps_front"
      boundingObject USE GPS_BOX
      physics Physics {
      }
    }
    GPS {
      translation -0.1 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
          }
          geometry DEF GPS_BOX Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "gps_back"
      boundingObject USE GPS_BOX
      physics Physics {
      }
    }
  ]
  name "R2-LED2"
  boundingObject USE BODY
  physics Physics {
  }
  controller "my_controller"
}
